### PR TITLE
Base theme manager setup

### LINKED ISSUE #12: Setup a theme manager
## Description

One thing that should be baked into the ui-library is dynamic color support. 
We should be able to change the colors programmatically based on user-supplied color hexes.

Think for example how the premium theme customization works in the rotki app currently.

## Tasks

- [ ] Figure out how popular vue3 component libraries handle themes
- [ ] Plan and implement some kind of theme manager

### PR DESCRIPTION
**Pull request by tewshi**
Closes #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,45 +41,50 @@
...
+    "@storybook/addon-essentials": "7.0.7",
+    "@storybook/addon-interactions": "7.0.7",
+    "@storybook/addon-links": "7.0.7",
+    "@storybook/addon-styling": "1.0.5",
+    "@storybook/blocks": "7.0.7",
+    "@storybook/testing-library": "0.1.0",
+    "@storybook/vue": "7.0.7",
+    "@storybook/vue-vite": "7.0.7",
+    "@types/node": "18.16.1",
+    "@vitejs/plugin-vue": "4.2.1",
```
Comment (kelsos) [inline package.json:56]: Is there any specific reason why we would need to move the devDepedencies to dependencies? 

The library should not depend directly on storybook we only use it for tooling so I would assume that the proper place would be devDependencies.

Comment (tewshi) [reply package.json:56]: I'll move them to dev, was having lots of challenges when building, so I moved them temporarily, will return the indirect dependencies to dev

File: .gitignore [hunk 2]
```diff
@@ -118,9 +118,8 @@ dist
 # vuepress build output
 .vuepress/dist
 
-# vuepress v2.x temp and cache directory
+# vuepress v2.x temp
 .temp
-.cache
```
Comment (kelsos) [inline .gitignore:123]: Any specific reason to remove these from .gitignore? 

I can't see why would we want a folder named `.cache` outside of the ignored directories.

Comment (tewshi) [reply .gitignore:123]: it was repeated, there's another on line 101

**Review by kelsos — APPROVED** [review]
lgtm

**Pull request merged**